File: app/src/main/java/me/wcy/music/service/OnPlayerEventListener.java
package me.wcy.music.service;

import me.wcy.music.model.Music;

/**
 * 播放进度监听器
 * Created by hzwangchenyan on 2015/12/17.
 */
public interface OnPlayerEventListener {

    /**
     * 切换歌曲
     */
    void onChange(Music music);

    /**
     * 继续播放
     */
    void onPlayerStart();

    /**
     * 暂停播放
     */
    void onPlayerPause();

    /**
     * 更新进度
     */
    void onPublish(int progress);

    /**
     * 缓冲百分比
     */
    void onBufferingUpdate(int percent);

    /**
     * 更新定时停止播放时间
     */
    void onTimer(long remain);

    void onMusicListUpdate();
}

File: app/src/main/java/me/wcy/music/service/OnPlayerEventListener.kt
package me.wcy.music.service

import me.wcy.music.model.Music

/**
 * 播放进度监听器
 * Created by hzwangchenyan on 2015/12/17.
 */
interface OnPlayerEventListener {
    /**
     * 切换歌曲
     */
    fun onChange(music: Music?)

    /**
     * 继续播放
     */
    fun onPlayerStart()

    /**
     * 暂停播放
     */
    fun onPlayerPause()

    /**
     * 更新进度
     */
    fun onPublish(progress: Int)

    /**
     * 缓冲百分比
     */
    fun onBufferingUpdate(percent: Int)
}